packages/core/src/node-execution-context/helpers/binary-helpers.ts

```diff
@@ -0,0 +1,148 @@
+import FileType from 'file-type';
+import { IncomingMessage } from 'http';
+import MimeTypes from 'mime-types';
+import { ApplicationError, fileTypeFromMimeType } from 'n8n-workflow';
+import type {
+	BinaryHelperFunctions,
+	IWorkflowExecuteAdditionalData,
+	Workflow,
+	IBinaryData,
+} from 'n8n-workflow';
+import path from 'path';
+import type { Readable } from 'stream';
+import Container from 'typedi';
+
+import { BinaryDataService } from '@/BinaryData/BinaryData.service';
+import { binaryToBuffer } from '@/BinaryData/utils';
+// eslint-disable-next-line import/no-cycle
+import { binaryToString } from '@/NodeExecuteFunctions';
+
+export class BinaryHelpers {
+	private readonly binaryDataService = Container.get(BinaryDataService);
+
+	constructor(
+		private readonly workflow: Workflow,
+		private readonly additionalData: IWorkflowExecuteAdditionalData,
+	) {}
+
+	get exported(): BinaryHelperFunctions {
+		return {
+			getBinaryPath: this.getBinaryPath.bind(this),
+			getBinaryMetadata: this.getBinaryMetadata.bind(this),
+			getBinaryStream: this.getBinaryStream.bind(this),
+			binaryToBuffer,
+			binaryToString,
+			prepareBinaryData: this.prepareBinaryData.bind(this),
+			setBinaryDataBuffer: this.setBinaryDataBuffer.bind(this),
+			copyBinaryFile: this.copyBinaryFile.bind(this),
+		};
```

Comment:
Should we perhaps define all the methods on this class as properties instead so we wouldn't need to do the binding and create new function instances?

I.e.
```ts
...
getBinaryPath = (binaryDataId: string) => {
  ...
}
...
```

Reply:
eventually, yes. but for now I want these PRs to make the least amount of changes possible.

Reply:
We actually currently also create new function instances for execution context. And, `this.helpers` is (unfortunately) not the same object across various execution context types.
I'd like to merge all the helpers into a single file, and make all the same helpers available across all context types. 
Created https://linear.app/n8n/issue/CAT-281 to do that later.